### PR TITLE
CSHARP-1875: Steer clear of code that needs ntdll.dll

### PR DESCRIPTION
**Pull request by MarLoe**
As few changes as possible to accomplish the same functionality without using those parts of System.Runtime.InteropServices.RuntimeInformation that will reference ntdll.dll

### CODE REVIEW

File: src/MongoDB.Driver.Core/Core/Connections/ClientDocumentHelper.cs [hunk 1]
```diff
@@ -69,44 +69,117 @@ internal static BsonDocument CreateDriverDocument(string driverVersion)
...
             string osVersion;
-            var match = Regex.Match(osName, @" (?<version>\d+\.\d[^ ]*)");
-            if (match.Success)
-            {
-                osVersion = match.Groups["version"].Value;
+
+#if NET45
+            if (Type.GetType("Mono.Runtime") != null)
+            {
+                Console.WriteLine("Hello Mono!!!");
```
Comment (derickr) [inline src/MongoDB.Driver.Core/Core/Connections/ClientDocumentHelper.cs]: Were you meant to leave this in here?

Comment (MarLoe) [inline src/MongoDB.Driver.Core/Core/Connections/ClientDocumentHelper.cs]: Absolutely not. I must have blinked... ;)

PR has been updated

**Comment by G3N7** [comment]
This is a blocker for any folks using Mono, any ETR for getting this merged in and released?

**Comment by tedblad** [comment]
Agree with @G3N7, this is a blocker. A merge and release would be very much appreciated.

**Comment by bondjames12** [comment]
Simple change and 1 less dependency. Fixes mono compatibility (even if its not officially tested)

**Comment by Zaid-Ajaj** [comment]
After succesfully setting up docker with mono, I am stuck with this problem. Merging any time soon?

**Comment by rstam** [comment]
It has been merged. See: https://jira.mongodb.org/browse/CSHARP-1875

We have not yet determined when the next release will be so it the change is not yet available on Nuget.

**Comment by Zaid-Ajaj** [comment]
@rstam Thanks for the quick response! For the people still blocked on this issue: just roll back MongoDb driver to 2.3.0 where there is no need for System.RuntimeInformation and delete all packages the depend on System.RuntimeInformation and it works (at least on my ubuntu 16.04 + mono 4.6.1.5 inside docker)

**Pull request closed without merging**